### PR TITLE
[Bug-10394] [Update Possword] Change password shows error

### LINKED ISSUE #10394: [Bug-Backend] [Update Possword] Change password shows error
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/dolphinscheduler/issues?q=is%3Aissue) and found no similar issues.

### What happened

Change password shows error

image

image

### What you expected to happen

Successfully modified

Back-end verification, as long as the old and new passwords are not equal

### How to reproduce

change Password
Before modification: dolphinscheduler123
Modified: dolphinscheduler123456

### Anything else

_No response_

### Version

3.0.0-beta-1

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by syyangs799**
…teraction

## Purpose of the pull request
This pull request add a status code for password verification length to improve user interaction 
Fixed  [ #10394 ](https://github.com/apache/dolphinscheduler/issues/10394#issue-1266902762) 

## Verify this pull request

This pull request is code cleanup without any test coverage.

*(or)*

This pull request is already covered by existing tests, such as *(please describe tests)*.

(or)

This change added tests and can be verified as follows:

### CODE REVIEW

File: dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/service/impl/UsersServiceImpl.java [hunk 1]
```diff
@@ -414,7 +414,7 @@ public Map<String, Object> updateUser(User loginUser, int userId,
 
         if (StringUtils.isNotEmpty(userPassword)) {
             if (!CheckUtils.checkPassword(userPassword)) {
-                putMsg(result, Status.REQUEST_PARAMS_NOT_VALID_ERROR, userPassword);
+                putMsg(result, Status.USER_PASSWORD_NOT_VALID_OR_TOO_LONG_ERROR, userPassword);
```
Comment (caishunfeng) [inline dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/service/impl/UsersServiceImpl.java]: please check separately and return different tips.

**Review by caishunfeng — CHANGES_REQUESTED** [review]

**Comment by caishunfeng** [comment]
Hi @syyangs799 please link the issue to this pr description, such as this pr close #issueId.

**Comment by syyangs799** [comment]
> Hi @syyangs799 please link the issue to this pr description, such as this pr close #issueId.

@caishunfeng ok I have modified it to different tips and added corresponding tests

**Comment by zhuangchong** [comment]
@syyangs799 Please resolve the conflicting code.

**Comment by syyangs799** [comment]
> @syyangs799 Please resolve the conflicting code.

Okay, I'm done

File: dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/service/impl/UsersServiceImpl.java [hunk 2]
```diff
@@ -413,10 +414,15 @@ public Map<String, Object> updateUser(User loginUser, int userId,
...
-            if (!CheckUtils.checkPassword(userPassword)) {
+
+            if (!StringUtils.isEmpty(userPassword)) {
                 putMsg(result, Status.REQUEST_PARAMS_NOT_VALID_ERROR, userPassword);
                 return result;
             }
+            if (CheckUtils.checkPasswordLength(userPassword)) {
+                putMsg(result, Status.USER_PASSWORD_LENGTH_ERROR);
+                return result;
+            }
```
Comment (caishunfeng) [inline dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/service/impl/UsersServiceImpl.java]:
```suggestion
            if (StringUtils.isEmpty(userPassword)) {
                putMsg(result, Status.REQUEST_PARAMS_NOT_VALID_ERROR, userPassword);
                return result;
            }
            if (CheckUtils.checkPasswordLength(userPassword)) {
                putMsg(result, Status.USER_PASSWORD_LENGTH_ERROR);
                return result;
            }
```

File: dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/service/impl/UsersServiceImpl.java [hunk 3]
```diff
@@ -413,10 +414,15 @@ public Map<String, Object> updateUser(User loginUser, int userId,
         }
 
         if (StringUtils.isNotEmpty(userPassword)) {
```
Comment (caishunfeng) [inline dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/service/impl/UsersServiceImpl.java:416]: remove the outer not empty validate

Comment (syyangs799) [reply dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/service/impl/UsersServiceImpl.java:416]: > remove the outer not empty validate

I found that my conditional judgment was wrong. I was very confused about the other judgment. I didn't rule out null values, but I didn't modify it to maintain consistency

Comment (caishunfeng) [reply dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/service/impl/UsersServiceImpl.java:416]: > > remove the outer not empty validate
> 
> I found that my conditional judgment was wrong. I was very confused about the other judgment. I didn't rule out null values, but I didn't modify it to maintain consistency

Sorry @syyangs799 , I found that my original comment is wrong, the outer non-null judgment needs to be retained, and will only be replaced when the new password is not null.
In this way, it's enough to keep the `checkPasswordLength` validate when password not empty, what do you think?

Comment (syyangs799) [reply dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/service/impl/UsersServiceImpl.java:416]: > > > remove the outer not empty validate
> > 
> > 
> > I found that my conditional judgment was wrong. I was very confused about the other judgment. I didn't rule out null values, but I didn't modify it to maintain consistency
> 
> Sorry @syyangs799 , I found that my original comment is wrong, the outer non-null judgment needs to be retained, and will only be replaced when the new password is not null. In this way, it's enough to keep the `checkPasswordLength` validate when password not empty, what do you think?

ok，I understand that this is to modify the user's other non password information

**Review by caishunfeng — APPROVED** [review]
LGTM :+1:

**Comment by syyangs799** [comment]
> LGTM 👍

@caishunfeng Thank you for my first PR  o(￣▽￣)ｄ

**Comment by syyangs799** [comment]
@caishunfeng Sorry, I clicked the wrong button

**Review by zhuangchong — APPROVED** [review]
LGTM.

**Comment by caishunfeng** [comment]
> > LGTM +1
> 
> @caishunfeng Thank you for my first PR o(￣▽￣)ｄ

Thanks for your contribution, looking forward to your next pr :smile:

**Pull request merged**